tests/ci/unit_tests_check.py

```diff
@@ -43,67 +45,120 @@ def get_test_name(line):
+    }"""
 
     test_results = []  # type: TestResults
-    total_counter = 0
-    failed_counter = 0
-    result_log_path = f"{result_folder}/test_result.txt"
-    if not os.path.exists(result_log_path):
-        logging.info("No output log on path %s", result_log_path)
-        return "error", "No output log", test_results, []
-
-    status = "success"
+    report_path = result_directory / "test_result.json"
+    if not report_path.exists():
+        logging.info("No output log on path %s", report_path)
+        return "error", "No output log", test_results
+
+    with open(report_path, "r", encoding="utf-8") as j:
+        report = json.load(j)
+
+    total_counter = report["tests"]
+    failed_counter = report["failures"]
+    error_counter = report["errors"]
+    status = SUCCESS
+    if failed_counter:
+        status = FAILURE
+    if error_counter:
+        status = ERROR
+
     description = ""
-    passed = False
-    with open(result_log_path, "r", encoding="utf-8") as test_result:
-        for line in test_result:
-            if OK_SIGN in line:
-                logging.info("Found ok line: '%s'", line)
-                test_name = get_test_name(line.strip())
-                logging.info("Test name: '%s'", test_name)
-                test_results.append(TestResult(test_name, "OK"))
-                total_counter += 1
-            elif FAILED_SIGN in line and "listed below" not in line and "ms)" in line:
-                logging.info("Found fail line: '%s'", line)
-                test_name = get_test_name(line.strip())
-                logging.info("Test name: '%s'", test_name)
-                test_results.append(TestResult(test_name, "FAIL"))
-                total_counter += 1
-                failed_counter += 1
-            elif SEGFAULT in line:
-                logging.info("Found segfault line: '%s'", line)
-                status = "failure"
-                description += "Segmentation fault. "
-                break
-            elif SIGNAL in line:
-                logging.info("Received signal line: '%s'", line)
-                status = "failure"
-                description += "Exit on signal. "
-                break
-            elif PASSED in line:
-                logging.info("PASSED record found: '%s'", line)
-                passed = True
-
-    if not passed:
-        status = "failure"
-        description += "PASSED record not found. "
-
-    if failed_counter != 0:
-        status = "failure"
+    SEGFAULT = "Segmentation fault. "
+    SIGNAL = "Exit on signal. "
+    for suite in report["testsuites"]:
+        suite_name = suite["name"]
+        for test_case in suite["testsuite"]:
+            case_name = test_case["name"]
+            test_time = float(test_case["time"][:-1])
+            raw_logs = None
+            if "failures" in test_case:
+                raw_logs = ""
+                for failure in test_case["failures"]:
+                    raw_logs += failure["failure"]
```

Comment:
Based on the protobuf above the `Failure` struct has a `failures` field, but not `failure`.
```suggestion
                    raw_logs += failure["failures"]
```